vpr/src/pack/prepack.cpp

```diff
@@ -1141,11 +1141,10 @@ static AtomBlockId get_sink_block(const AtomBlockId block_id,
     // Iterate through all sink blocks and check whether any of them
     // is compatible with the block specified in the pack pattern.
     bool connected_to_latch = false;
-    LogicalModelId latch_model_id = LogicalModels::MODEL_LATCH_ID;
     AtomBlockId pattern_sink_block_id = AtomBlockId::INVALID();
     for (const auto& sink_pin_id : net_sinks) {
         auto sink_block_id = atom_nlist.pin_block(sink_pin_id);
-        if (atom_nlist.block_model(sink_block_id) == latch_model_id) {
+        if (!atom_nlist.block_is_combinational(sink_block_id)) {
```

Comment:
I think it would be better to add a comment explaining why this API call is used (we’re assuming that if a block has a clock, it’s stateful and therefore falls into the category we want to detect).